lib/xclarity_client/xclarity_base.rb

```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
     def connection_builder(conf, uri)
       $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Creating connection to #{conf.host + uri}" 
       #Building configuration
-      @conn = Faraday.new(url: conf.host + uri) do |faraday|
+      host = conf.host;
+      if !host.start_with?('https://') && !host.start_with?('http://')
+         host = 'https://' + host
```

Comment:
I just added a commit to refactor how this is implemented.   With the refactor if any protocol is specified we will use the specified protocol ( instead of just http and https).  If no protocol is specified we will assume https and use the port details provided.   I believe that MIQ is passing down port details when it constructus the provider connection.